### PR TITLE
treq: 17.3.1 fixes

### PR DESCRIPTION
**Pull request by nand0p**
- enables and fixes tests
- makes docs properly
- tested nixos python 27/34

### CODE REVIEW

File: pkgs/development/python-modules/treq/default.nix [hunk 1]
```diff
@@ -33,13 +33,12 @@ buildPythonPackage rec {
...
-  # Failure: twisted.web._newclient.RequestTransmissionFailed: [<twisted.python.failure.Failure OpenSSL.SSL.Error: [('SSL routines', 'ssl3_get_server_certificate', 'certificate verify failed')]>]
+  patchPhase = ''
+    rm -fv src/treq/test/test_treq_integration.py
+  '';
 
   postBuild = ''
-    ${coreutils}/bin/mkdir -pv treq
-    ${coreutils}/bin/echo "${version}" | ${coreutils}/bin/tee treq/_version
-    cd docs && ${gnumake}/bin/make html && cd ..
+    tox -e docs
```
Comment (FRidh) [inline pkgs/development/python-modules/treq/default.nix]: what is this for?

File: pkgs/development/python-modules/treq/default.nix [hunk 2]
```diff
@@ -33,13 +33,12 @@ buildPythonPackage rec {
     trial treq
   '';
 
-  doCheck = false;
-  # Failure: twisted.web._newclient.RequestTransmissionFailed: [<twisted.python.failure.Failure OpenSSL.SSL.Error: [('SSL routines', 'ssl3_get_server_certificate', 'certificate verify failed')]>]
+  patchPhase = ''
```
Comment (FRidh) [inline pkgs/development/python-modules/treq/default.nix]: Please always use `postPatch` when adding additional patching. That way, one can still pass in `patches`.

File: pkgs/development/python-modules/treq/default.nix (hunk 1 above)
Comment (nand0p) [inline pkgs/development/python-modules/treq/default.nix]: makes documentation available in $out

Comment (FRidh) [inline pkgs/development/python-modules/treq/default.nix]: do always document why you run commands like these because its not obvious at all what it does.

**Comment by nand0p** [comment]
@FRidh ready_for_review

File: pkgs/development/python-modules/treq/default.nix [hunk 3]
```diff
@@ -26,20 +20,27 @@ buildPythonPackage rec {
...
+    rm -fv src/treq/test/test_treq_integration.py
+  '';
 
   postBuild = ''
-    ${coreutils}/bin/mkdir -pv treq
-    ${coreutils}/bin/echo "${version}" | ${coreutils}/bin/tee treq/_version
-    cd docs && ${gnumake}/bin/make html && cd ..
+    # build documentation and install in $out
+    tox -e docs
+    cp -rv docs $out
```
Comment (FRidh) [inline pkgs/development/python-modules/treq/default.nix]: Should they not go in a specific folder? Note that with Python packages we typically do not bother with the docs.

**Comment by nand0p** [comment]
@FRidh updated to correctly write out documentation to `$out/docs`. i left this in for the sake of completeness, but let me know if you would rather remove documentation all-together.  it seems if the author took the time to write it, we should attempt to package, if not too much trouble?

**Comment by FRidh** [comment]
@nand0p its just that its the wrong destination. Documentation likely goes somewhere in `/share`. I'm not too familiar with those things though.

**Comment by nand0p** [comment]
@FRidh im seeing other python mods like `python-wifi`, write out documentation to `$out/docs`, but im also seeing other projects write out docs to `$out/share` and `$out/lib`.   im not finding any documentation regarding what the correct method is.  can we merge this for now, so as to unblock master, and I can update once there is a clear path forward regarding documentation?

**Pull request merged**